### ISSUE TITLE
"Value 'Optional' is unsubscriptable" on Python 3.9

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#3861]
### Steps to reproduce

Run pylint on the following code:

```py
from typing import Optional

x: Optional[int]
```

### Current behavior

```
testcase.py:3:3: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No error is reported on that line.

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.9.0rc1 (default, Aug 18 2020, 12:19:39) [GCC]
```

The same version of pylint and astroid on Python 3.8.5 does not report any error, so the problem is triggered by a change in Python 3.9.

**Comment by fniessink** [comment]
This is probably a duplicate report of #3760.

**Comment by mthuurne** [comment]
That does seem likely. I'll add the test case there and close this issue.
